x/multisig/types/testutils/rand.go

```diff
@@ -40,6 +44,45 @@ func Key() types.Key {
 	}
 }
 
+// KeyWithMissingParticipants returns a random multisig key with some missing participants
+func KeyWithMissingParticipants() types.Key {
+	participantCount := uint64(rand.I64Between(10, 20))
+	missingCount := uint64(rand.I64Between(1, int64(participantCount)))
+	participants := slices.Expand(func(_ int) snapshottypes.Participant {
+		return snapshottypes.Participant{Address: rand.ValAddr(), Weight: sdk.NewUint(uint64(rand.I64Between(1, 100)))}
+	}, int(participantCount))
+
+	missingParticipants := slices.Expand(func(_ int) snapshottypes.Participant {
+		return snapshottypes.Participant{Address: rand.ValAddr(), Weight: sdk.NewUint(uint64(rand.I64Between(1, 100)))}
+	}, int(missingCount))
+
+	participants = append(participants, missingParticipants...)
+	weightAdder := func(total sdk.Uint, p snapshottypes.Participant) sdk.Uint { total = total.Add(p.Weight); return total }
+	participantWeight := slices.Reduce(participants, sdk.ZeroUint(), weightAdder)
+	missingParticipantWeight := slices.Reduce(missingParticipants, sdk.ZeroUint(), weightAdder)
+
+	bondedWeight := rand.UintBetween(participantWeight, participantWeight.MulUint64(2))
+	threshold := utils.NewThreshold(rand.I64Between(1, participantWeight.Sub(missingParticipantWeight).BigInt().Int64()), bondedWeight.BigInt().Int64())
```

Comment:
Threshold generation is easier to reason about in this manner when participants are missing, instead of reusing the existing snapshot creation function.